features/kiesrecht.feature

```diff
@@ -108,46 +108,235 @@ Rule: Uitsluiting van Europees kiesrecht wordt alleen opgenomen wanneer de eindd
 		| einddatum uitsluiting onbekend     | false              | 00000000              | false             | DatumOnbekend  |            |                |                 |
 		| persoon ontvangt oproep            | true               |                       | true              |                |            |                |                 |
 
-@proxy
-Rule: Uitsluiting van kiesrecht wordt alleen opgenomen wanneer de einddatum uitsluiting in de toekomst ligt
-	- Wanneer alleen het jaar van de einddatum uitsluiting bekend is, dan wordt de uitsluiting opgenomen tot en met dat jaar.
-	- Wanneer het jaar en de maand van de einddatum uitsluiting bekend is, en de dag niet, dan wordt de uitsluiting opgenomen tot en met die maand.
-	- Een volledig onbekende einddatum uitsluiting wordt hetzelfde geïnterpreteerd en weergegeven als het niet aanwezig zijn van die datum.
-	- Wanneer einddatum uitsluiting geen waarde heeft en uitsluiting kiesrecht wel, wordt uitsluiting kiesrecht wel opgenomen.
-	- Wanneer einddatum uitsluiting in het verleden ligt of vandaag is, wordt het niet opgenomen.
+    Scenario: persoon heeft geen uitsluiting van kiesrecht
```

Comment:
hier bedoel je uitsluiting van Europees kiesrecht neem ik aan?

Reply:
is dit scenario niet hetzelfde als het eerste voorbeeld van het vorige abstracte scenario?

Reply:
nee, ik bedoel hier kiesrecht, maar eigenlijk geldt dit ook voor Europees kiesrecht, want in deze scenario heeft de persoon geen kiesrecht gegevens

Reply:
> is dit scenario niet hetzelfde als het eerste voorbeeld van het vorige abstracte scenario?

Het hangt af hoe die eerste voorbeeld moet worden geïnterpreteerd. Wat wordt met die eerste voorbeeld gezegd? Ik kom op de volgende drie opties:

```
{
  "burgerservicenummer": "..."
}
```
of
```
{
  "burgerservicenummer": "...",
  "kiesrecht": {}
}
```
of
```
{
  "burgerservicenummer": "...",
  "kiesrecht": {
     "uitgeslotenVanKiesrecht": "",
    "einddatumUitsluitingKiesrecht": ""
  }
}
```
waarvan de laatste is wat cucumber maakt uit een table. Verder zijn er hierdoor nu 2 manieren om aan te geven dat een gegevensgroep niet is gevuld. Als voorbeeld met kiesrecht:
- Gegeven de persoon heeft GEEN 'kiesrecht' gegevens
- Gegeven de persoon heeft de volgende 'kiesrecht' gegevens
  | naam | waarde|
  | uitgeslotenVanKiesrecht | |
  | einddatumUitsluitingKiesrecht | |

Niet echt handig, want dit soort voor toename van automation code. Mijn voorkeur gaat daarom naar één manier, de eerste. Maar persoonlijk zou ik helemaal geen 'Gegeven' stap gebruiken om aan te geven dat een persoon een gegevensgroep niet heeft. Er wordt in de scenario's ook niet expliciet gezegd dat een persoon geen geboorte, nationaliteiten, partners, etc heeft

Reply:
vanuit automation perspectief heeft het inderdaad geen zin te zeggen dat iets er niet is. iets is er pas wanneer er een stap is die zegt dat het er is.
vanuit beschrijven van de functionaliteit kan dat m.i. anders zijn. soms wil je expliciet aangeven dat iets er niet is, omdat dit aanduidt welke situatie getoond en getest wordt.

Wanneer de automation code werkelijk "uitgeslotenVanKiesrecht": "" zou sturen, zou dat een foutmelding moeten geven, want het is een string i.p.v. een boolean. Volgens mij hebben we het hier al eerder over gehad en wordt ook in andere features een lege waarde in een kolom bedoeld als niet aanwezig, niet als lege string. 
Bijvoorbeeld aanschrijfwijze.feature regel 613:
```
      | naam                          | waarde |
      | aanschrijfwijze.aanspreekvorm |        |
```
En in dezelfde feature tabel op regel 335 in voorvoegsel.

Reply:
Dit scenario "persoon heeft geen uitsluiting van kiesrecht" staat onder de rule "Rule: Uitsluiting van Europees kiesrecht wordt alleen opgenomen wanneer ...", maar spreekt in de titel over uitsluiting van kiesrecht. Ook zit het voorbeeld met de lege waarden nog in de tabel, op regel 98. Vandaar mijn verwarring. 
Moet dit scenario niet onder rule "Rule: Uitsluiting van kiesrecht wordt opgenomen wanneer ..." worden geplaatst?

En moet dan regel 98 niet uit de tabel worden verwijderd, en in plaats daarvan een scenario "persoon heeft geen Europees kiesrecht" worden toegevoegd